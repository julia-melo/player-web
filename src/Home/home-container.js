import React from 'react'

import HomeView from './home-view'

import mock from './tests/mock'

const HomeContainer = () => (
  <HomeView videoInfo={mock} recommendations={mock.recommendations}/>
)

export default HomeContainer
