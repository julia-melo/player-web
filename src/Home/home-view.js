import React from 'react'
import PropTypes from 'prop-types'

import Player from '../components/Player'
import Track from '../components/Track'
import Thumb from '../components/Thumb'

import './index.css'

const HomeView = ({ videoInfo, recommendations }) => (
  <div className='home'>
    <Player {...videoInfo} />

    <div className='recommendations'>
      <Track itensList={recommendations}>
        {(item) => (
          <Thumb
            image={item.thumb}
            size={160}
            caption={item.title}
          />
        )}
      </Track>
    </div>
  </div>
)

HomeView.propTypes = {
  videoInfo: PropTypes.exact({
    id: PropTypes.number,
    title: PropTypes.string,
    thumb: PropTypes.string,
  }),
  recommendations: PropTypes.array,
}

export default HomeView
