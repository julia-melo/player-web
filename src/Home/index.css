.home {
  width: 100vw;
  height: 100vh;

  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.home .recommendations {
  width: 800px;
}

.home .player {
  margin-bottom: 40px;
}
