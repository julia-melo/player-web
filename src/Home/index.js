import React from 'react'

import Player from '../components/Player'
import Track from '../components/Track'
import Thumb from '../components/Thumb'

import mock from './mock'
import './index.css'

const Home = () => (
  <div className='home'>
    <Player {...mock} />

    <div className='recommendations'>
      <Track itensList={mock.recommendations}>
        {(item) => (
          <Thumb
            image={item.thumb}
            size={160}
            caption={item.title}
          />
        )}
      </Track>
    </div>
  </div>
)

export default Home
