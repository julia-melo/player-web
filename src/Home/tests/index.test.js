import React from 'react'
import renderer from 'react-test-renderer'

import Home from '../home-view'
import mock from './mock'

describe('Home component', () => {
  it('renders correctly', () => {
    const videoInfoMock = {
      id: mock.id,
      title: mock.title,
      thumb: mock.thumb
    }

    const tree = renderer
      .create(<Home videoInfo={videoInfoMock} recommendations={mock.recommendations} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})