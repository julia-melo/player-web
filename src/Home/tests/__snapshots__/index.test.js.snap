// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  className="home"
>
  <div
    className="player"
  >
    <p
      className="title"
    >
      Você não vai acreditar no que esse gatinho fez
    </p>
    <div
      className="content"
      onClick={[Function]}
    >
      <div
        className="thumb"
      >
        <div
          className="content"
          style={Object {}}
        >
          <img
            alt="thumb"
            className="image"
            src="https://i.ytimg.com/vi/dGFSjKuJfrI/maxresdefault.jpg"
          />
          <div
            className="play-icon"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="recommendations"
  >
    <div
      className="track"
    >
      <div
        className="arrow-left-icon disabled "
      />
      <div
        className="track-item"
      >
        <div
          className="thumb"
        >
          <div
            className="content"
            style={
              Object {
                "height": "100px",
                "width": "160px",
              }
            }
          >
            <img
              alt="thumb"
              className="image"
              src="https://petgusto.com/wp-content/uploads/2017/08/gatos-fofos-e-engracados-1.jpg"
            />
            <div
              className="play-icon"
            />
          </div>
          <p
            className="title"
            style={
              Object {
                "width": "160px",
              }
            }
          >
            Gatinhos encontram câmera escondida
          </p>
        </div>
      </div>
      <div
        className="track-item"
      >
        <div
          className="thumb"
        >
          <div
            className="content"
            style={
              Object {
                "height": "100px",
                "width": "160px",
              }
            }
          >
            <img
              alt="thumb"
              className="image"
              src="https://img.vixdata.io/pd/jpg-large/pt/sites/default/files/g/gato-notebook-internet-0817-1400x800.jpg"
            />
            <div
              className="play-icon"
            />
          </div>
          <p
            className="title"
            style={
              Object {
                "width": "160px",
              }
            }
          >
            Home office com gatinhos
          </p>
        </div>
      </div>
      <div
        className="track-item"
      >
        <div
          className="thumb"
        >
          <div
            className="content"
            style={
              Object {
                "height": "100px",
                "width": "160px",
              }
            }
          >
            <img
              alt="thumb"
              className="image"
              src="https://i.ytimg.com/vi/9kkIzivnC78/maxresdefault.jpg"
            />
            <div
              className="play-icon"
            />
          </div>
          <p
            className="title"
            style={
              Object {
                "width": "160px",
              }
            }
          >
            Top 10 momentos de gatinhos na internet
          </p>
        </div>
      </div>
      <div
        className="track-item"
      >
        <div
          className="thumb"
        >
          <div
            className="content"
            style={
              Object {
                "height": "100px",
                "width": "160px",
              }
            }
          >
            <img
              alt="thumb"
              className="image"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSW53eTqgSobpTAHa3I1C2XOTHFdJkFTA5Keg&usqp=CAU"
            />
            <div
              className="play-icon"
            />
          </div>
          <p
            className="title"
            style={
              Object {
                "width": "160px",
              }
            }
          >
            Conheça o gatinho mais fofo do universo
          </p>
        </div>
      </div>
      <div
        className="arrow-right-icon  "
      />
    </div>
  </div>
</div>
`;
